Fix mtd resetbc

asd333111 opened the pull request:
1. Fix a bug that mtd cannot reset boot count when page size is larger than 2048
2. Add code to check the checksum in the boot-count log entry

testuser7 commented:
Can you rebase and update this PR?
Also commit title need to be changed and `Signed-off-by` message with full name added.

Commit:
mtd: fix resetbc on nand w/ min I/O size > 2048
Fix a bug in linksys_bootcount.c that resetbc  won't work on nand
with min I/O size> 2048.

Check the boot-log entry's intergrity with checksum.

Signed-off-by: Qiyuan Zhang <zhang.github@outlook.com>

asd111333 commented:
Rebased and updated.

asd333111 commented:
Updated
